lib/upipe-speexdsp/upipe_speexdsp.c

```diff
@@ -325,16 +362,12 @@ static int upipe_speexdsp_control(struct upipe *upipe, int command, va_list args
         /* generic commands */
         case UPIPE_REGISTER_REQUEST: {
             struct urequest *request = va_arg(args, struct urequest *);
-            if (request->type == UREQUEST_UBUF_MGR ||
-                request->type == UREQUEST_FLOW_FORMAT)
-                return upipe_throw_provide_request(upipe, request);
+            if (request->type == UREQUEST_FLOW_FORMAT)
+                return upipe_speexdsp_provide_flow_format(upipe, request);
```

Comment:
You still must handle UREQUEST_UBUF_MGR because you are technically a sink for ubufs.